File: RestFront/src/app/models/menu.ts
export class Menu {
    id: number
    name: string
    price: number
    restId: number
    constructor (
        id: number,
        name: string,
        price: number,
        restId: number
    ) {
        this.id = id;
        this.name = name;
        this.price = price;
        this.restId = restId;
    }
}

File: Resturaunt/ResturantAPI/ResturantAPI/Models/Menu.cs
using System.ComponentModel.DataAnnotations;

namespace ResturantAPI.Models;
public class Menu
{
    [Key]
    public int? Id { get; set; }
    public string? Name { get; set; }
    public decimal? Price { get; set; }
    public int? RestId { get; set; }
}